m2m/tag_mapper.py

```diff
@@ -24,6 +25,23 @@ def to_markdown(self):
             return self.markdown_unordered_list()
         elif self.tag.name == 'ol':
             return self.markdown_ordered_list()
+        elif self.tag.name == 'div':
+            return self.parse_div(self.tag)
+
+    def parse_div(self, div_tag):
+        for child in div_tag:
+            if child.name == 'div':
+                self.s += self.parse_div(child) if self.parse_div(child) else ''
```

Comment:
Does it do duplicate recursion here?

1. `self.s += self.parse_div(child)`
2. `if self.parse_div(child)`

Isn't it better to `parse_div` one time, "store" the value in a variable, and then use it? Something like this:

```python
div_string = self.parse_div(child)
self.s += div_string if div_string else ''
```

We can memoize the value here.